evalml/tests/component_tests/test_polynomial_detrender.py

```diff
@@ -96,17 +96,17 @@ def test_polynomial_detrender_inverse_transform(degree, use_int_index, ts_data):
     pd.testing.assert_series_equal(y, output_inverse_y, check_dtype=False)
 
 
-def test_polynomial_detrender_needs_monotonic_index(ts_data):
+def test_polynomial_detrender_needs_monotonic_index(ts_data, has_minimal_dependencies):
     X, y = ts_data
     detrender = PolynomialDetrender(degree=2)
 
     with pytest.raises(ValueError) as exec_info:
         y_shuffled = y.sample(frac=1, replace=False)
         detrender.fit_transform(X, y_shuffled)
     assert "monotonically" in str(exec_info.value)
-
+    error = ValueError if not has_minimal_dependencies else NotImplementedError
     with pytest.raises(
-        NotImplementedError,
+        error,
```

Comment:
sktime v0.10.0 raises a `ValueError` here, but our `min_dependencies` test raises a `NotImplementedError` instead since it uses sktime v0.7.0